service/src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -416,7 +448,55 @@ public void removeWorkspaceRole(
   }
 
   /**
-   * Wrapper around Sam client to retrieve the current permissions model of a workspace.
+   * Wrapper around the Sam client to remove a role from the provided user on a controlled resource.
+   *
+   * <p>Similar to {@removeWorkspaceRole}, but for controlled resources. This should only be
+   * necessary for private resources, as users do not have individual roles on shared resources.
+   *
+   * <p>This call to Sam is made as the WSM SA, as users do not have permission to directly modify
+   * IAM on resources. This method still requires user credentials to validate as a safeguard, but
+   * they are not used in the role removal call.
+   */
+  @Traced
```

Comment:
It would help to document the params here.

Reply:
done